gliderpy/plotting.py

```diff
@@ -90,3 +90,51 @@ def plot_transect(
     ax.set_ylim(ax.get_ylim()[0], 0)
 
     return fig, ax
+
+
+@register_dataframe_method
+def plot_ctd(
+    df: pd.DataFrame,
+    profile_number: int,
+    var: str,
+    ax: plt.Axes = None,
+    color: str | None = None,
+) -> tuple:
+    """Make a CTD profile plot of pressure vs property
+    depending on what variable was chosen.
+
+    :param profile_number: profile number of CTD
+    :param var: variable to plot against pressure
+    :param ax: existing axis to plot on (default: None)
+    :param color: color for the plot line (default: None)
+    :return: figure, axes
+    """
+    g = df.groupby(["longitude", "latitude"])
+    profile = g.get_group(list(g.groups)[profile_number])
+
+    if ax is None:
+        fig, ax = plt.subplots(figsize=(5, 6))
+        ax.plot(profile[var], profile["pressure"], label=var, color=color)
+        ax.set_ylabel("Pressure")
+        ax.set_xlabel(var)
+        ax.legend()
+        ax.invert_yaxis()
+        return fig, ax
+
+    fig = ax.get_figure()
+
+    # Check if the ax is already a twinx or twiny
+    if hasattr(ax, "twinned_axis"):
+        ax = ax.twinned_axis
+    else:
+        ax = ax.twiny()
+        ax.twinned_axis = ax  # Keep a reference to the twinned axis
+
+    ax.plot(profile[var], profile["pressure"], label=var, color=color)
+    ax.set_xlabel(var)
+
+    lines, labels = ax.get_legend_handles_labels()
+    lines2, labels2 = ax.get_legend_handles_labels()
+    ax.legend(lines + lines2, labels + labels2, loc="lower center")
```

Comment:
We don't need this now that we have a single `ax`.

```suggestion
```
```suggestion
```

Reply:
The pre-commit stopped running on my computer. I don't know why, so I couldn't review the errors. I will try again later.